### PR TITLE
Assorted docs fixes

### PR DESCRIPTION
**Pull request by alanbernstein**
## Overview

This contains a bunch of docs fixes and additions, focusing on "docs landing pages", i.e., introduction, getting started, readme, and other pages explicitly linked from those. Fixes are focused on syntax (to ensure all code snippets in the docs run as expected) and wording (to remove outdated references, and to make every subsection as clear and self-contained as possible).

High level summary:
- Added a "relational analogy" section. We can expand this, but I wanted to get some basic feedback first.
- Ran every code snippet on the included pages, fixed syntax errors, displayed results in a "response" block (query-language, examples, client-libraries).
- Picked a consistent format for query-language sections. The query description comes before the "request" code block, which is now followed by a "response" block, and the query description is written first in terms of rows/columns/bits (then repositories and stargazers in parens). This page was previously written under the assumption that a user would read it straight through, and be fully aware of the stargazer/repository context. The sections need to make sense to a new user who arrives on the page through an anchor link to read about one specific query, without referring back to the setup.
- Consolidated rowAttr and columnAttr information, with more explicit info about retrieving columnAttrs.
- Most of the other changes are wording tweaks

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: docs/client-libraries.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ nav = [
 
 ## Client Libraries
 
-This section contains example code for client libraries in several languages. Please remember that when modeling your data in Pilosa, it is best to keep row and column ids sequential. It is not wise to use the output of a hash, or randomly distributed ids with Pilosa.
+This section contains example code for client libraries in several languages. Please remember that when modeling your data in Pilosa, it is best to keep row and column ids sequential. It is best to avoid using the the output of a hash, or randomly distributed ids with Pilosa.
```
Comment (travisturner) [inline docs/client-libraries.md]: the comma after "hash" is not necessary, unless you also put a comma after "ids".

File: docs/data-model.md [hunk 2]
```diff
@@ -20,36 +20,74 @@ nav = [
...
+Rows and columns can represent anything (they could even represent the same set of things - a [bigraph](https://en.wikipedia.org/wiki/Bigraph)). Pilosa can associate arbitrary key/value pairs (referred to as attributes) to rows and columns, but queries and storage are optimized around the core matrix.
 
-Pilosa lays out data first in rows, so queries which get all the set bits in one or many rows, or compute a combining operation on multiple rows such as Intersect or Union are the fastest. Pilosa also has the ability to categorize rows into different "frames" and quickly retrieve the top rows in a frame sorted by the number of bits set in each row.
+Pilosa lays out data first in rows, so queries which get all the set bits in one or many rows, or compute a combining operation on multiple rows such as Intersect or Union are the fastest. Pilosa categorizes rows into different *frames* and quickly retrieve the top rows in a frame sorted by the number of bits set in each row.
```
Comment (travisturner) [inline docs/data-model.md]: retrieve > retrieves

File: docs/data-model.md [hunk 3]
```diff
@@ -20,36 +20,74 @@ nav = [
...
+Column ids are sequential increasing integers and are common to all Frames within an Index. A single column often corresponds to a record in a relational table, although other configurations are possible, and sometimes preferable (see genomics TODO).
 
 ### Row
 
 Row ids are sequential increasing integers namespaced to each Frame within an Index.
 
 ### Frame
 
-Frames are used to segment and define different functional characteristics within your entire index.  You can think of a Frame as a table-like data partition within your Index.
+Frames are used to segment rows within an index, for example to defining different functional groups. A frame might correspond to a single field in a relational table, where each row in a standard frame represents a single possible value of the field. Similarly, a frame with BSI values could represent all possible integer values of a field .
```
Comment (travisturner) [inline docs/data-model.md]: just noting that "index" is lowercase here, but capitalized in other places. 
"to defining" > "to define"

File: docs/data-model.md [hunk 4]
```diff
@@ -97,12 +137,11 @@ SetBit(frame="A", row=8, col=3, timestamp="2017-05-19T00:00")
 
 #### BSI Range-Encoding
 
-Bit-Sliced Indexing (BSI) is the storage method Pilosa uses to represent multi-bit integers in a bitmap index. Integers are stored as n-bit, range-encoded
-bit-sliced indexes of base-2, along with an additional bitmap indicating "not null". This means that a 16-bit integer will require 17 bitmaps: one for each 0-bit of the 16 bit-slice components (the 1-bit does not need to be stored because with range-encoding the highest bit position is always 1) and one for the non-null bitmap. Pilosa can evaluate `Range`, `Min`, `Max`, and `Sum` queries on these BSI integers.
+Bit-Sliced Indexing (BSI) is the storage method Pilosa uses to represent multi-bit integers in a bitmap index. Integers are stored as n-bit, range-encoded bit-sliced indexes of base-2, along with an additional bitmap indicating "not null". This means that a 16-bit integer will require 17 bitmaps: one for each 0-bit of the 16 bit-slice components (the 1-bit does not need to be stored because with range-encoding the highest bit position is always 1) and one for the non-null bitmap. Pilosa can evaluate `Range`, `Min`, `Max`, and `Sum` queries on these BSI integers.
```
Comment (travisturner) [inline docs/data-model.md]: in addition to "Sum" should we include "Avg" as a derived result?

File: docs/examples.md [hunk 5]
```diff
@@ -197,12 +203,16 @@ for pcount, topn in zip(pcounts, resp.json()['results']):
     average_amounts.append(float(wsum)/count)
 ```
 
+<div class="note">
+Note that the <a href="../data-model/#bsi-range-encoding">BSI</a>-powered <a href="../query-language/#sum">Sum</a> query now provides a an alternative approach to this kind of query.
```
Comment (travisturner) [inline docs/examples.md]: "a an" > "an"

File: docs/data-model.md (hunk 3 above)
Comment (alanbernstein) [reply docs/data-model.md]: We aren't very consistent with styling Pilosa technical terms. In the white paper some are lowercase and italicized, others are wrapped in quotation marks. In the docs we have the option to link to a relevant section (glossary for example). We don't have any existing guidelines for this, do we?

File: docs/faq.md [hunk 6]
```diff
@@ -16,7 +16,7 @@ Pilosa is not a database in the traditional sense. While Pilosa does store data
 
 ### Where does Pilosa fit in my stack?
 
-Pilosa sits on top of a data store or multiple data stores.
+Pilosa was designed to index the relationships in your data. Pilosa runs along with your existing stack, integrating with one or more backing data stores. Pilosa can connect through a stream platform like Kafka or application integration via [PDK](../pdk/).
```
Comment (alanbernstein) [inline docs/faq.md]: @codysoyland @skingkong how does this sound?

Comment (codysoyland) [reply docs/faq.md]: Sounds good! 👍

**Pull request merged**